Formula/knot-resolver.rb

```diff
@@ -0,0 +1,101 @@
+class KnotResolver < Formula
+  desc "Minimalistic, caching, DNSSEC-validating DNS resolver"
+  homepage "https://www.knot-resolver.cz"
+  url "https://secure.nic.cz/files/knot-resolver/knot-resolver-1.2.1.tar.xz"
+  sha256 "1b6f55ea1dfec90f45c437f23e1ab440e478570498161d0f8a8f94a439305f8c"
+  revision 1
+
+  head do
+    url "https://gitlab.labs.nic.cz/knot/resolver.git"
+  end
+
+  depends_on "knot"
+  depends_on "luajit"
+  depends_on "libuv"
+  depends_on "gnutls"
+  depends_on "lmdb"
+
+  depends_on "cmocka" => :build
+  depends_on "pkg-config" => :run
+
+  option "without-nettle", "Compile without DNS cookies support"
+  option "with-hiredis", "Compile with Redis cache storage support"
+  option "with-libmemcached", "Compile with memcached cache storage support"
+  depends_on "nettle" => :recommended
+  depends_on "hiredis" => :optional
+  depends_on "libmemcached" => :optional
+
+  def install
+    %w[all check lib-install daemon-install modules-install].each do |target|
+      system "make", target, "PREFIX=#{prefix}"
+    end
+
+    (buildpath/"config").write(config)
+    (etc/"kresd").install "config"
+
+    (buildpath/"root.keys").write(root_keys)
+    (var/"kresd").install "root.keys"
+  end
+
+  def config; <<-EOS.undent
+    -- vim:syntax=lua:
+    -- Refer to manual: http://knot-resolver.readthedocs.org/en/latest/daemon.html#configuration
+
+    net.listen(net.lo0)
+
+    -- Unmanaged DNSSEC root TA
+    trust_anchors.file = 'root.keys'
+
+    -- Load useful modules
+    modules = {
+        'stats',    -- Track internal statistics
+        'policy',   -- Block queries to local zones/bad sites
+        'predict',  -- Prefetch expiring/frequent records
+    }
+
+    -- Cache size
+    cache.size = 100 * MB
+    EOS
+  end
+
+  # DNSSEC root anchor published by IANA
+  def root_keys; <<-EOS.undent
```

Comment:
@MikeMcQuaid @vcunat There is still some confusion in this discussion.

The file `root_keys` is no sample or configuration file. It's secure trust anchor for DNSSEC. The anchor rarely changes. First key was published in Jun 2010 and the new one in Feb 2017. Don't expect that we would have to change the file frequently.

Also, the trust anchor is *not* installed by `make install`. The `iccan-ca.pem` certificate is installed instead which is something different. In case `root_keys` is not provided, kresd will try to bootstrap this file by downloading the anchor from IANA website and by using the ICANN certificate to verify it's content. This may or may not work.

I understand the concern of _harcoding_ something into the formula. But I still think we should follow what for instance Linux distributions do and provide `root_keys` in the _package_. This is most comfortable for users. And also a robust solution because we can be sure that they will always have the trust anchor.